### PR TITLE
CI/CD migration

### PR DESCRIPTION
**Pull request by jmarcosfer**
### Changes:
- Travis config has been removed
- Added an integration workflow that uses Github Actions to build the library on every push or PR open/edit to branches dev or master
- Added an example deploy workflow (which, for now, just temporarily uploads the builds to [transfer.sh](https://transfer.sh/)) to run on PR closed on dev and master

### Discussion:
- Unsure about [cpp setup](https://github.com/MTG/essentia.js/blob/b07e84f1e11f90293988414de4b19503400ec19e/.github/workflows/integration.yml#L47) to mimick `language: cpp` line on travis.yml file. Commented it for now, since I assume cpp setup is already done in docker image.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - master
+    paths:
+      - 'src/**'
+      # - 'test/**'
+      - 'build-libs.sh'
+      - 'Dockerfile'
+      - 'Makefile.essentiajs'
+      - 'rollup*.js'
+      - 'tsconfig.json'
+      - '!docs/**'
```
Comment (albincorreya) [inline .github/workflows/deploy.yml:16]: We don't need `docs` and `examples` directory here, right?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  build-deploy:
+    runs-on: ubuntu-latest
+    container:
+      image: docker://mtgupf/essentia-emscripten:latest
+      volumes: 
+        - ${{ env.GITHUB_WORKSPACE }}:/essentia/
+      options: --rm
+    steps:
+      - uses: actions/checkout@v2
+      # - uses: kurocha/setup-cpp@v1 # necessary?
```
Comment (albincorreya) [inline .github/workflows/deploy.yml:31]: remove this line

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: docker://mtgupf/essentia-emscripten:latest
+      volumes: 
+        - ${{ env.GITHUB_WORKSPACE }}:/essentia/
+      options: --rm
+    steps:
+      - uses: actions/checkout@v2
+      # - uses: kurocha/setup-cpp@v1 # necessary?
```
Comment (albincorreya) [inline .github/workflows/integration.yml:47]: remove this line

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      - master
+    paths:
+      - 'src/**'
+      # - 'test/**'
+      - 'build-libs.sh'
+      - 'Dockerfile'
+      - 'Makefile.essentiajs'
+      - 'rollup*.js'
+      - 'tsconfig.json'
+      - '!docs/**'
```
Comment (albincorreya) [inline .github/workflows/integration.yml:30]: same comment as above

File: .github/workflows/integration.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    runs-on: ubuntu-latest
+    container:
+      image: docker://mtgupf/essentia-emscripten:latest
+      volumes: 
+        - ${{ env.GITHUB_WORKSPACE }}:/essentia/
+      options: --rm
+    steps:
+      - uses: actions/checkout@v2
+      # - uses: kurocha/setup-cpp@v1 # necessary?
+      - run: /essentia/build-libs.sh
```
Comment (albincorreya) [inline .github/workflows/integration.yml:48]: in future we can split this bash script to run build and tests separately. Not important in this PR.

**Review by albincorreya — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (albincorreya) [reply .github/workflows/deploy.yml:31]: since we are running it inside docker this is not necessary.

**Comment by albincorreya** [comment]
LGTM @jmarcosfer. Btw, were you able to run successful job in actions with the current branch. Checking [here](https://github.com/MTG/essentia.js/actions) seems like the last job was failed.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jmarcosfer) [reply .github/workflows/deploy.yml:16]: Exactly, the `/!docs/**` and `!examples/**` lines are excluding PRs that only change docs or examples, so it should be good as it is

File: .github/workflows/integration.yml (hunk 4 above)
Comment (jmarcosfer) [reply .github/workflows/integration.yml:30]: yep, again, the `!` excludes those paths

File: .github/workflows/integration.yml (hunk 5 above)
Comment (jmarcosfer) [reply .github/workflows/integration.yml:48]: agree, tests should eventually be a separate job, although in that case we need to find out how to share data between jobs (we will need the output of the build job for the test job)

**Comment by jmarcosfer** [comment]
> LGTM @jmarcosfer. Btw, were you able to run successful job in actions with the current branch. Checking [here](https://github.com/MTG/essentia.js/actions) seems like the last job was failed.

Yes, the failed job is from the first time I pushed, the path for one of the `uses` fields was wrong, but should be corrected already. In any case, I'll try running these workflows locally with https://github.com/nektos/act and confirm that everything's working fine. Then I'll comment here and close the PR. Thanks for the review!

**Review by albincorreya — APPROVED** [review]

**Comment by albincorreya** [comment]
@jmarcosfer One last thing, it would be nice to replace the 'build passing' badge in README with github action status instead of travis. Perhaps [this](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) could be a way of doing it. I guess, there might be lot of alternatives out there.

**Comment by jmarcosfer** [comment]
Eventually the `deploy.yml` workflow (which currently just builds the project like the integration workflow does and puts it on [transfer.sh](https://transfer.sh/)) could be setup to publish the library on the npm site and CDNs triggered by a release event on `master`.

**Comment by albincorreya** [comment]
> Eventually the deploy.yml workflow (which currently just builds the project like the integration workflow does and puts it on transfer.sh) could be setup to publish the library on the npm site and CDNs triggered by a release event on master.

@jmarcosfer Yes, I agree. It would be nice setup that too. I see you already have some some in `ci` directory. Perhaps we can also add that script to the `deploy.yml` workflow.

Regarding, publishing to NPM can be setup as it only works with a manual trigger step in the pipeline (no automatic push).

**Pull request merged**